src/function/gds/strongly_connected_components.cpp

```diff
@@ -0,0 +1,346 @@
+        bindData = std::make_unique<GDSBindData>(std::move(graphEntry), nodeOutput);
+    }
+
+    void exec(processor::ExecutionContext* context) override {
+        auto clientContext = context->clientContext;
+        auto mm = clientContext->getMemoryManager();
+        auto graph = sharedState->graph.get();
+        auto getMaxOffsetMap = graph->getMaxOffsetMap(clientContext->getTransaction());
+        auto it = getMaxOffsetMap.begin();
+        auto tableId = it->first;
+        auto numNodes = it->second;
+
+        auto computationState = SCCComputationState(numNodes, mm);
+        // Initialize SCC IDs to unset.
+        auto setInitialSccIds = std::make_unique<SetInitialSccIds>(computationState);
+        GDSUtils::runVertexCompute(context, graph, *setInitialSccIds);
+
+        // The frontiers will be initialized inside the loop.
+        auto currentFrontier = std::make_shared<PathLengths>(getMaxOffsetMap, mm);
+        auto nextFrontier = std::make_shared<PathLengths>(getMaxOffsetMap, mm);
+        currentFrontier->pinCurFrontierTableID(tableId);
+        nextFrontier->pinCurFrontierTableID(tableId);
+        auto frontierPair =
+            std::make_unique<DoublePathLengthsFrontierPair>(currentFrontier, nextFrontier);
+        frontierPair->getCurSparseFrontier().disable();
+        frontierPair->getNextSparseFrontier().disable();
+
+        auto setNewSccIds = std::make_unique<FindNewSccIds>(computationState);
+        auto setInitialColors = std::make_unique<SetInitialColors>(computationState);
+        auto computeColors = std::make_unique<ComputeColors>(computationState);
+        auto auxiliaryState = std::make_unique<SccAuxiliaryState>();
+        auto computeState = GDSComputeState(std::move(frontierPair), std::move(computeColors),
+            std::move(auxiliaryState), sharedState->getOutputNodeMaskMap());
+        auto initializeFrontiers =
+            std::make_unique<InitializeFrontiers>(*computeState.frontierPair, computationState);
+
+        for (auto i = 0; i < MAX_ITERATION; i++) {
+            GDSUtils::runVertexCompute(context, graph, *setInitialColors);
+
+            // Fwd colors.
+            computeState.frontierPair->initState();
+            computeState.frontierPair->initGDS();
+            GDSUtils::runVertexCompute(context, graph, *initializeFrontiers);
+            GDSUtils::runFrontiersUntilConvergence(context, computeState, graph,
+                ExtendDirection::FWD, MAX_ITERATION);
+
+            // Bwd colors.
+            computeState.frontierPair->initState();
+            computeState.frontierPair->initGDS();
+            GDSUtils::runVertexCompute(context, graph, *initializeFrontiers);
+            GDSUtils::runFrontiersUntilConvergence(context, computeState, graph,
+                ExtendDirection::BWD, MAX_ITERATION);
+
+            // Find new SCC IDs and exit if all IDs have been found.
+            setNewSccIds->reset();
```

Comment:
Instead of reset from a vertex compute function. I would directly reset the SCCComputationState.